dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/RouterChain.java

```diff
@@ -167,21 +149,12 @@ public void addStateRouters(List<StateRouter<T>> stateRouters) {
         BitList<Invoker<T>> resultInvokers = availableInvokers.clone();
 
         // 1. route state router
-        for (StateRouter<T> stateRouter : stateRouters) {
-            StateRouterResult<Invoker<T>> routeResult = stateRouter.route(resultInvokers, url, invocation, false);
-            resultInvokers = routeResult.getResult();
-            if (resultInvokers.isEmpty() && shouldFailFast) {
-                printRouterSnapshot(url, availableInvokers, invocation);
-                return BitList.emptyList();
-            }
-
-            // stop continue routing
-            if (!routeResult.isNeedContinueRoute()) {
-                return routeResult.getResult();
-            }
+        resultInvokers = headStateRouter.route(resultInvokers, url, invocation, false, null);
+        if (resultInvokers.isEmpty() && shouldFailFast) {
+            printRouterSnapshot(url, availableInvokers, invocation);
+            return BitList.emptyList();
         }
 
-
         if (routers.isEmpty()) {
```

Comment:
if resultInvokers and routers both is empty,and it will not print router snapshot